javaslang/src/main/java/javaslang/collection/List.java

```diff
@@ -557,13 +557,19 @@
 
     @Override
     default List<T> append(T element) {
-        return foldRight(of(element), (x, xs) -> xs.prepend(x));
+        final ArrayList<T> results = new ArrayList<>(size() + 1);
```

Comment:
Please do the List changes in a separate PR. This PR should contain only benchmark changes.